python/lsst/ts/externalscripts/auxtel/latiss_acquire.py

```diff
@@ -0,0 +1,577 @@
+                self.object_ra,
+                self.object_dec,
+                target_name=self.object_name,
+                rot=self.rot_value,
+                rot_type=getattr(RotType, self.rot_type),
+                slew_timeout=240,
+                az_wrap_strategy=self.azimuth_wrap_strategy,
+                time_on_target=self.time_on_target,
+            )
+        # Otherwise use the target name to query Simbad for coordinates.
+        else:
+            self.log.debug("Using slew_object (object name designation)")
+            _slew_coro = self.atcs.slew_object(
+                name=self.object_name,
+                rot=self.rot_value,
+                rot_type=getattr(RotType, self.rot_type),
+                slew_timeout=240,
+                az_wrap_strategy=self.azimuth_wrap_strategy,
+                time_on_target=self.time_on_target,
+            )
+
+        tmp, data = await asyncio.gather(
+            _slew_coro,
+            self.latiss.setup_atspec(grating=self.acq_grating, filter=self.acq_filter),
+        )
+
+        # Perform acquisition loop to center target.
+        await self.execute_acquisition(self.target_position)
+
+        # Verify with another image that we're on target?
+        if self.target_pointing_verification:
+            await self.latiss.take_acq(
+                exptime=self.acq_exposure_time,
+                n=1,
+                group_id=self.group_id,
+                reason=self.reason,
+                program=self.program,
+            )
+        else:
+            self.log.debug(
+                f"Skipping additional image to verify offset was applied correctly as "
+                f"target_pointing_verification is set to {self.target_pointing_verification}"
+            )
+
+    async def latiss_reacquire(self):
+        await self.latiss.setup_atspec(grating=self.acq_grating, filter=self.acq_filter)
+
+        await self.execute_acquisition(self.target_position)
+
+        # Verify with another image that we're on target?
+        if self.target_pointing_verification:
+            await self.latiss.take_acq(
+                exptime=self.acq_exposure_time,
+                n=1,
+                group_id=self.group_id,
+                reason=self.reason,
+                program=self.program,
+            )
+        else:
+            self.log.debug(
+                f"Skipping additional image to verify offset was applied correctly as "
+                f"target_pointing_verification is set to {self.target_pointing_verification}"
+            )
+
+    async def assert_feasibility(self) -> None:
+        """Verify that the telescope and camera are in a feasible state to
+        execute the script.
+        """
+
+        await self.atcs.assert_all_enabled()
+        await self.latiss.assert_all_enabled()
+
+        self.log.debug("Check ATAOS corrections are enabled.")
+        ataos_corrections = await self.atcs.rem.ataos.evt_correctionEnabled.aget(
```

Comment:
We should really move this to the `ATCS` class...

If you don't want to do it now, can you add a TODO and issue a ticket?

Reply:
https://jira.lsstcorp.org/browse/DM-38823